### PR TITLE
Implement Connext xERC20 and Circle FiatToken collateral

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Implement XERC20 Collateral contract
- Implement FiatToken Collateral contract

### Drive-by

- Move vault collateral extension into token extensions folder

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3486

### Backward compatibility

- Yes

### Testing

- Unit tests

### CODE REVIEW

**Comment by avious00** [comment]
@nambrot @tkporter pls review

**Comment by yorhodes** [comment]
I dont think we need to pull @tkporter into this
also not ready for review yet

File: typescript/sdk/src/token/TokenStandard.ts [hunk 1]
```diff
@@ -53,6 +55,8 @@ export const TOKEN_STANDARD_TO_PROTOCOL: Record<TokenStandard, ProtocolType> = {
   EvmHypCollateral: ProtocolType.Ethereum,
   EvmHypcollateralVault: ProtocolType.Ethereum,
   EvmHypSynthetic: ProtocolType.Ethereum,
+  EvmHypXERC20Collateral: ProtocolType.Ethereum,
+  EvmHypFiatCollateral: ProtocolType.Ethereum,
```
Comment (jmrossy) [inline typescript/sdk/src/token/TokenStandard.ts:59]: I think you'll need to hook these new standards up to a token adapter in `Token.ts#getHypAdapter`

File: typescript/sdk/src/token/deploy.ts [hunk 2]
```diff
@@ -67,23 +66,7 @@ export class HypERC20Deployer extends GasRouterDeployer<
...
-      }
-    } else if (isNativeConfig(config)) {
-      return config.scale ? TokenType.nativeScaled : TokenType.native;
-    } else if (isSyntheticConfig(config)) {
-      return isFastConfig(config)
-        ? TokenType.fastSynthetic
-        : TokenType.synthetic;
-    } else {
-      throw new Error('Unknown collateral type when constructing router name');
-    }
```
Comment (jmrossy) [inline typescript/sdk/src/token/deploy.ts:86]: Nice.

**Review by jmrossy — APPROVED** [review]
LGTM

**Review by nbayindirli — APPROVED** [review]
🚢

**Pull request merged**